src/pages/index.js

```diff
@@ -17,47 +42,111 @@ popupImage.setEventListeners();
 const openPopupImage = (data) => {
   popupImage.open(data);
 };
-const renderCard = (card) => {
-  cardRenderer.addItem(createCard(card));
+
+const deletePopup = new PopupWithForm(constants.deletePopup, (data) => {
+  api
+    .deleteCard(data._id, data)
+    .then(() => {
+      deletePopup.submitButton.textContent = "Delete Card";
+      cardArray = document.querySelectorAll(".card");
+      cardArray.forEach((card) => {
+        if (card.id === data._id) {
+          card.remove();
+        }
+      });
+      deletePopup.close();
+    })
+    .catch((error) => {
+      alert(`Request to server failed. ${error}`);
+    });
+});
+deletePopup.setEventListeners();
+
+const openDeletePopup = (data) => {
+  deletePopup.open();
+  constants.deleteID.value = data._id;
 };
-const cardRenderer = new Section(
-  { items: constants.initialCards, renderer: renderCard },
-  constants.cardList
-);
 
-cardRenderer.renderItems();
+function sendLike(data) {
+  if (data.isLiked) {
+    api
+      .removeLike(data._id, data)
+      .then((data.isLiked = false))
+      .then(alert(`Like for "${data.name}" removed`));
+  } else {
+    api
+      .likeCard(data._id, data)
+      .then((data.isLiked = true))
+      .then(alert(`Like for "${data.name}" added`));
+  }
+}
 
 const userInfo = new UserInfo({
   name: constants.profileName,
-  description: constants.profileDesc,
+  about: constants.profileDesc,
 });
 const cardModal = new PopupWithForm(constants.cardModal, (data) => {
-  cardRenderer.addItem(createCard(data));
+  api
+    .sendCard(data)
+
+    .then((result) => {
+      cardRenderer.addItem(createCard(result));
+      cardModal.submitButton.textContent = "Create";
+      cardModal.close();
+    });
 });
+
 cardModal.setEventListeners();
+const avatarModal = new PopupWithForm(constants.changeAvatar, (data) => {
+  api.updateProfilePicture(data).then((result) => {
+    constants.avatar.src = result.avatar;
 
-const profileModal = new PopupWithForm(constants.profileModal, (data) => {
-  userInfo.setUserInfo(data);
+    avatarModal.close();
+    avatarModal.submitButton.textContent = "Change Picture";
+  });
 });
-profileModal.setEventListeners();
+avatarModal.setEventListeners();
 
 // Card Code
 
 function createCard(card) {
-  const initCard = new Card(card, "#card", openPopupImage);
+  const initCard = new Card(
+    card,
+    "#card",
+    openPopupImage,
+    openDeletePopup,
+    sendLike
+  );
   return initCard.createCard();
 }
 
 //Modal Box Code
-
+const profileModal = new PopupWithForm(constants.profileModal, (data) => {
+  api.updateUserInfo(data);
+  constants.profileName.textContent = data.name;
+  constants.profileDesc.textContent = data.about;
+  profileModal.close();
```

Comment:
You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well

**Please, correct it everywhere**